### PR TITLE
Logging gradients with tensor's names

### PR DESCRIPTION
**Pull request by MartinPernus**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?   Not needed
- [ ] Did you write any new necessary tests?  Not needed

## What does this PR do?
Current gradient logging (when enabled) provides no information about the names of the parameters of interest, providing only raw values of gradient norms. With PyTorch's named_parameters() method we can access the names of our parameters - this is now used to name the individual gradient norm.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by williamFalcon** [comment]
@MartinPernus thanks!
